pandas/_libs/intervaltree.pxi.in

```diff
@@ -342,6 +342,13 @@ cdef class {{dtype_title}}Closed{{closed_title}}IntervalNode(IntervalNode):
             # calculate a pivot so we can create child nodes
             self.is_leaf_node = False
             self.pivot = np.median(left / 2 + right / 2)
+            if np.isinf(self.pivot):
```

Comment:
Could `self.pivot` be `np.nan` here at some point?

Reply:
That would be the case if one of the interval boundaries would be `np.nan`. I think is hard to trigger if you don't use the private `IntervalTree` class directly. Doing something like `Interval(left=0.0, right=np.nan)` is caught by the validator which enforces  `left side of interval must be <= right side`.

If you do use `IntervalTree()` to force a `np.nan` interval boundary (which is IMO nonsensical), then `self.pivot` would indeed become `np.nan`. But even that case does not trigger the infinite recursion. Only nonsensical input → nonsensical output.

So to conclude I would think there is no need to catch `np.isnan(self.pivot)`.

Reply:
Okay great, thanks for the explaination.